lib/inputs/file-pattern/static.ts

```diff
@@ -0,0 +1,64 @@
+import * as minimatch from "minimatch";
+import * as path from "path";
+
+import { ExtractAction, ExtractedLayers } from "../../extractor/types";
+import { streamToString } from "../../stream-utils";
+import { ManifestFile } from "../../types";
+
+function generatePathMatcher(
+  globsInclude: string[],
+  globsExclude: string[],
+): (filePath: string) => boolean {
+  return (filePath: string): boolean => {
+    let exclude = false;
+    for (const g of globsExclude) {
+      if (!exclude && minimatch(filePath, g)) {
+        exclude = true;
+      }
+    }
+    if (!exclude) {
+      for (const g of globsInclude) {
+        if (minimatch(filePath, g)) {
+          return true;
+        }
+      }
+    }
+    return false;
+  };
+}
+
+export function generateExtractAction(
+  globsInclude: string[],
+  globsExclude: string[],
+): ExtractAction {
+  return {
+    actionName: "find-files-by-pattern",
+    filePathMatches: generatePathMatcher(globsInclude, globsExclude),
+    callback: streamToString,
+  };
+}
+
+export function getMatchingFiles(
+  extractedLayers: ExtractedLayers,
+): ManifestFile[] {
```

Comment:
it's totally unclear if I'm going for a generic retrieve file mechanism or specific to manifest files